packages/fuels-types/src/errors.rs

```diff
@@ -47,20 +45,18 @@ pub enum Error {
     Utf8Error(#[from] Utf8Error),
     #[error("Compilation error: {0}")]
     CompilationError(String),
-    #[error("Network error: {0}")]
-    NetworkError(String),
```

Comment:
NetworkError I'm not sure should disappear, what happens if for instance the node actually works, but your localhost has badly configured firewall or whatever? It's semantically very different IMO.

Reply:
I looked into this a bit more. The gql client always returns a generic error type of variant [Other](https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html#variant.Other) which means we cannot differentiate between actual network errors or other client related errors by parsing the error type. I prefer to return a generic `ClientRequestError` over parsing error messages.